lib/charms/mongodb_libs/v0/mongodb_tls.py

```diff
@@ -66,13 +66,15 @@ def _on_set_tls_private_key(self, event: ActionEvent) -> None:
         logger.debug("Request to set TLS private key received.")
         try:
             self._request_certificate("unit", event.params.get("external-key", None))
+            self.charm.set_secret("unit", "cert", None)
```

Comment:
Please move this line into the `_request_certificate` call.
the same for line 77